src/generators/TypeScriptGenerator.ts

```diff
@@ -97,81 +117,178 @@ async function generateInterfaceFile(
+  );
 
-  // Transform the EntityDefinition for use as an interface type
-  const interfaceTransformer = new EntityInterfaceTransformer(baseDirectory);
-  const interfaceEntityDefinition = interfaceTransformer.transform(
-    entityDefinition
+  // Trim above list if necessary (not likely for real modelinfo)
+  const classesToHoist = _.intersection(hoistedClassNames, classTypeNames);
+
+  // Remove existing occurrences of above types
+  const trimmedClasses = _.difference(classTypeNames, classesToHoist);
+
+  // Add the above names to the front of the array
+  const classNames: Array<string> = [...classesToHoist, ...trimmedClasses];
+
+  // Get a list of all interface names
+  const interfaceTransformer = new TypeScriptInterfaceTransformer(
+    entityCollection.baseDir
+  );
+  const interfaceEntities = entityCollection.transform(interfaceTransformer);
+  const interfaceTypeNames: Array<string> = interfaceEntities.entities.map(
+    (entity) => entity.dataType.normalizedName
   );
 
-  await generateInterfaceFile(interfaceEntityDefinition, baseDirectory);
-
-  // If the interface type has a primitive, we need to import the IExtension interface
-  // const hasPrimitive = !!entityDefinition.memberVariables.find(
-  //   (memberVar) => memberVar.dataType.primitive
-  // );
-  // if (
-  //   hasPrimitive &&
-  //   !(namespace === "FHIR" && normalizedName === "Extension")
-  // ) {
-  //   const extensionType = DataType.getInstance("FHIR", "Extension", classPath);
-  //
-  //   const distinctTypes = safeAddTypeImport(
-  //     entityDefinition.imports.dataTypes,
-  //     extensionType,
-  //     entityDefinition.dataType
-  //   );
-  //
-  //   interfaceEntityDefinition = new EntityDefinition(
-  //     entityDefinition.metadata,
-  //     entityDefinition.dataType,
-  //     entityDefinition.parentDataType,
-  //     entityDefinition.memberVariables,
-  //     new EntityImports(distinctTypes)
-  //   );
-  // }
-
-  // If there are no members, or it's a primitive, just create an alias
-  // if (hasNoMembers || isPrimitive) {
-
-  // const isElementType = namespace === "FHIR" && baseTypeName === "Element";
-  // if (
-  //   isPrimitive && isElementType
-  // ) {
-  //   const mappedValue =
-  //     primitiveTypeJsonMapping[
-  //       interfaceEntityDefinition.metadata.originalTypeName
-  //     ];
-  //
-  //   aliasType.isRootType = true;
-  //   aliasType.baseTypeName = mappedValue;
-  // }
-  // }
-
-  return "done"; // FIXME
+  // Trim above list if necessary (not likely for real modelinfo)
+  const interfacesToHoist = _.intersection(
+    hoistedInterfaceNames,
+    interfaceTypeNames
+  );
+
+  // Remove existing occurrences of above types
+  const trimmedInterfaces = _.difference(interfaceTypeNames, interfacesToHoist);
+
+  // Add the above names to the front of the array
+  const interfaceNames: Array<string> = [
+    ...interfacesToHoist,
+    ...trimmedInterfaces,
+  ];
+
+  const internalContents: string = internalTemplate({
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5923420)